### PR TITLE
Fixed #25466 -- Added backwards compatibility aliases for LoaderOrigin and StringOrigin.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/25466

### CODE REVIEW

**Comment by prestontimmons** [comment]
This makes sense. Too bad there isn't a way to raise a proper deprecation warning.

One small fix is needed. ``StringOrigin`` was previously importable from ``django.template``:

https://docs.djangoproject.com/en/1.7/ref/templates/api/#template-origin

It was removed from ``django/template/__init__.py`` in fc2147152637.

**Comment by charettes** [comment]
What about hooking into `__instancecheck__` to raise the warning?

We did something similar with [`EmptyQuerySet`](https://github.com/django/django/blob/134ca4d438bd7cbe8f0f287a00d545f96fa04a01/django/db/models/query.py#L1171-L1183).

Since the `type` would be used in two modules we could add another helper to `django.utils.deprecation`. e.g.

```python
class DeprecationInstanceCheck(type):
    def __instancecheck__(self, instance):
        warnings.warn(
            "%s is deprecated, use %s instead." % (self.__name__, self.alternative),
            self.deprecation_warning, 2
        )
        return super(DeprecationInstanceCheck, self).__instancecheck__(instance)

class StringOrigin(six.with_metaclass(DeprecationInstanceCheck, Origin))
    alternative = 'django.template.Origin'
    deprecation_warning = RemovedInDjango20Warning
```

**Comment by timgraham** [comment]
Updated for comments.

File: tests/deprecation/tests.py [hunk 1]
```diff
@@ -170,3 +172,14 @@ class Deprecated(DeprecatedMixin, RenamedMixin, Renamed):
...
+
+class DeprecationInstanceCheckTest(SimpleTestCase):
+    def test_warning(self):
+        class Manager(six.with_metaclass(DeprecationInstanceCheck)):
+            alternative = 'fake.path.Foo'
+            deprecation_warning = RemovedInNextVersionWarning
+
+        msg = '`Manager` is deprecated, use `fake.path.Foo` instead.'
+        with self.assertRaisesMessage(RemovedInNextVersionWarning, msg):
+            isinstance(object, Manager)
```
Comment (charettes) [inline tests/deprecation/tests.py]: Didn't know `__instancecheck__` was never called when `instance.__class__ is self`.

**Comment by charettes** [comment]
LGTM

**Pull request merged**